.github/workflows/build.yaml

```diff
@@ -3,12 +3,25 @@ name: Build
 on: [push, pull_request]
 
 jobs:
-  build:
+  test:
     runs-on: ubuntu-latest
-    env:
-      # disable the demon, as it's causing some problems
-      GRADLE_OPTS: -Dorg.gradle.daemon=false
+    strategy:
+      matrix:
+        version: [ 8, 9, 11, 17 ]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
```

Comment:
I think we want to use ` ${{ matrix.version }}` here?

Reply:
We could, but gradle plays catch-up with every new version of java. By using the matrix version of java with the toolchain we can use unsupported versions of gradle (like 17 for example) in gradle 6 for compilation/testing.

Reply:
I don't mind either way, but it does mean testing on java 17 would require gradle 7.

Which shouldn't be too hard, considering we're now on gradle 6.

Reply:
I'm fine with skipping Java 17 for now.